external/CMakeLists.txt

```diff
@@ -115,60 +202,117 @@ set(BREAKPAD_SOURCES_CLIENT_IOS
+        endif()
 
     include(CheckFunctionExists)
     check_function_exists(getcontext HAVE_GETCONTEXT)
     if(HAVE_GETCONTEXT)
-        target_compile_definitions(breakpad_client PRIVATE HAVE_GETCONTEXT)
-    else()
-        target_sources(breakpad_client PRIVATE ${BREAKPAD_SOURCES_COMMON_LINUX_GETCONTEXT})
+        target_compile_definitions(breakpad_common PUBLIC HAVE_GETCONTEXT)
+        target_sources(breakpad_common PRIVATE ${BREAKPAD_SOURCES_COMMON_LINUX_GETCONTEXT})
     endif()
-
-    set_property(TARGET breakpad_client PROPERTY POSITION_INDEPENDENT_CODE ON)
 endif()
 
 if(APPLE)
-    target_sources(breakpad_client PRIVATE
-        ${BREAKPAD_SOURCES_COMMON_APPLE}
-        ${BREAKPAD_SOURCES_CLIENT_APPLE})
+    target_sources(breakpad_common PRIVATE ${BREAKPAD_SOURCES_COMMON_APPLE})
+    target_sources(breakpad_common_extra PRIVATE ${BREAKPAD_SOURCES_COMMON_EXTRA_UNIX} ${BREAKPAD_SOURCES_COMMON_EXTRA_APPLE})
+    target_sources(breakpad_client PRIVATE ${BREAKPAD_SOURCES_CLIENT_APPLE})
+    target_sources(dump_syms PRIVATE ${BREAKPAD_SOURCES_DUMPSYMS_MAC})
+    add_executable(macho_dump ${BREAKPAD_SOURCES_MACHODUMP_MAC})
+    target_link_libraries(macho_dump PRIVATE breakpad_common breakpad_common_extra)
+        target_compile_definitions(breakpad_common PRIVATE HAVE_MACH_O_NLIST_H)
+        target_compile_definitions(breakpad_common_extra PRIVATE HAVE_MACH_O_NLIST_H)
     if(NOT IOS)
-        target_sources(breakpad_client PRIVATE
-            ${BREAKPAD_SOURCES_COMMON_MAC}
-            ${BREAKPAD_SOURCES_CLIENT_MAC})
+        target_sources(breakpad_common PRIVATE ${BREAKPAD_SOURCES_COMMON_MAC})
+        target_sources(breakpad_client PRIVATE ${BREAKPAD_SOURCES_CLIENT_MAC})
     else()
-        target_sources(breakpad_client PRIVATE
-            ${BREAKPAD_SOURCES_CLIENT_IOS})
+        target_sources(breakpad_client PRIVATE ${BREAKPAD_SOURCES_CLIENT_IOS})
     endif()
 
-    target_link_libraries(breakpad_client PRIVATE "-framework CoreFoundation")
+    target_link_libraries(breakpad_common PRIVATE "-framework CoreFoundation")
 endif()
 
 if(WIN32)
-    target_sources(breakpad_client PRIVATE ${BREAKPAD_SOURCES_COMMON_WINDOWS} ${BREAKPAD_SOURCES_CLIENT_WINDOWS})
-    target_compile_definitions(breakpad_client PRIVATE _UNICODE UNICODE)
+    target_sources(breakpad_common PRIVATE ${BREAKPAD_SOURCES_COMMON_WINDOWS})
+    target_sources(breakpad_common_extra PRIVATE ${BREAKPAD_SOURCES_COMMON_EXTRA_WINDOWS})
+    target_sources(breakpad_client PRIVATE ${BREAKPAD_SOURCES_CLIENT_WINDOWS})
+    target_sources(dump_syms PRIVATE ${BREAKPAD_SOURCES_DUMPSYMS_WINDOWS})
+
+    target_compile_definitions(breakpad_common PUBLIC _UNICODE UNICODE)
 
-    # set static runtime if enabled
-    if(SENTRY_BUILD_RUNTIMESTATIC AND MSVC)
-        set_property(TARGET breakpad_client PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+    target_compile_options(breakpad_common_extra PUBLIC
+        "-I$(VsInstallDir)/DIA SDK/include"
```

Comment:
With MSVC, the *DIA SDK* library `diaguids` can be found in an awkward location.
The usage of `$(VSInstallDir)` is MSBuild only, I think.
Perhaps this deserves a find_package cmake module.

Also, I haven't tested this yet with the `NMake Makefiles` generator or with mingw.